content/blogs/2024-03-04-kestra-databricks.md

```diff
@@ -0,0 +1,159 @@
+
+The following code snippet creates the Databricks Compute cluster.
+
+```yaml
+id: databricks-create-cluster
+namespace: dev
+tasks:
+  - id: "create-cluster"
+    type: "io.kestra.plugin.databricks.cluster.CreateCluster"
+    authentication:
+    token: <your-token>
+    host: <your-host>
+    clusterName: kestra-demo
+    nodeTypeId: Standard_DS3_v2
+    numWorkers: 1
+    sparkVersion: 13.0.x-scala2.12
+```
+
+Post running this flow, you can check that the newly created compute cluster should now appear under the Compute section on the Databricks console.
+
+![create_compute_cluster](/blogs/2024-03-04-kestra-databricks/create_compute_cluster.png)
+
+### Upload file to Databricks File System ###
+
+Now that the compute cluster is running, let us upload a file to DBFS. Using the [upload task](https://kestra.io/plugins/tasks/dbfs/io.kestra.plugin.databricks.dbfs.upload) of the Kestra’s Databricks plugin, we can upload a file from Kestra’s internal storage onto DBFS. There is no restriction on the size of the file. The task will upload the file in chunks of 1 MB.
+
+The following code snippet will first download a csv file from HTTP URL onto Kestra’s internal storage. This file from Kestra’s internal storage will then be uploaded onto DBFS on the path specified in the `to` attribute of the Databricks' Upload task.
+
+```yaml
+id: databricks-upload
+namespace: dev
+tasks:
+  - id: http_download
+    type: io.kestra.plugin.fs.http.Download
+    uri: "https://raw.githubusercontent.com/kestra-io/datasets/main/csv/orders.csv"
+  - id: uploadFile
+    type: io.kestra.plugin.databricks.dbfs.Upload
+    authentication:
+    token: <your-token>
+    host: <your-host>
+    from: "{{ outputs.http_download.uri }}"
+    to: /Shared/kestra/data/orders.csv
+```
+
+Once the flow is successfully run, you can open the DBFS Browser on your Databricks console, and ensure that the file has got uploaded on the appropriate path.
+
+### Running Query on Databricks ###
+
+Let us now perform a couple of query operations on Databricks. Kestra’s Databricks plugin provides the [query task](https://kestra.io/plugins/tasks/sql/io.kestra.plugin.databricks.sql.query) for querying data from Databricks. This task requires you to mention the `httpPath`. In order to get the `httpPath`, you need to go to the compute cluster using which you need to perform this query. On the compute cluster’s page, go to `JDBC/ODBC` tab under the Advanced options. This is where you will get the `HTTP Path` which you can use in the `httpPath` attribute of your query task.
+
+The first query task that we will perform will be to load the data from the DBFS file onto a table. The query in this task will read the CSV file from the DBFS. It will create the table in the Databricks by inferring the schema from the CSV file, and the data will be filled in the table from this CSV file.
```

Comment:
```suggestion
The first query task that we will perform will be to load the data from the DBFS file into a table. The query in this task will read the CSV file from the DBFS. It will create the table in Databricks by inferring the schema from the CSV file, and the data will be filled in the table from this CSV file.
```